src/main/java/org/openlmis/stockmanagement/repository/mapper/LotMapper.java

```diff
@@ -15,7 +16,81 @@
       @Result(
           property = "product", column = "productId", javaType = Product.class,
           one = @One(select = "org.openlmis.core.repository.mapper.ProductMapper.getById")),
-      @Result(property = "lotCode", column = "lotnumber"),
+      @Result(property = "lotCode", column = "lotnumber")
   })
   Lot getById(@Param("id")Long id);
+
+  @Select("SELECT *" +
+      " FROM lots" +
+      " WHERE lotnumber = #{lotCode}" +
```

Comment:
Since lotCode and manufacturerName are text, check for case-insensitive equality